### PR TITLE
Feat/backend

### PR DESCRIPTION
**Pull request by tenzin3**

### CODE REVIEW

File: backend/backend_main.py [hunk 1]
```diff
@@ -0,0 +1,110 @@
...
+    - Answer directly, without adding any extra words.
+    - Be careful of the language, ensuring it is respectful and appropriate.
+    - If you do not have a proper answer from the context, respond with "I dont have enough data to provide an answer."
+    - Do not give a response longer than 3000 tokens.
+    
+    
+    Contexts: {context}
+    Question: {question}
+    
+    """
```
Comment (teny19) [inline backend/backend_main.py]: Put question first before providing the context

File: backend/backend_main.py [hunk 2]
```diff
@@ -0,0 +1,110 @@
...
+
+    qa_template = PromptTemplate(template)
+    prompt = qa_template.format(context_str=context, question_str=question)
```
Comment (teny19) [inline backend/backend_main.py]: should be question and not question_str

File: backend/backend_main.py [hunk 3]
```diff
@@ -0,0 +1,110 @@
...
+    
+    messages = [
+        {"role": "user", "content": prompt},
+    ]
+    # output = pipe(messages, **TEXT_GENERATION_ARGS)
+    # answer = output[0]['generated_text']
+    return context[:200]
```
Comment (teny19) [inline backend/backend_main.py]: why do we cut off at 200?

Comment (tenzin3) [reply backend/backend_main.py]: model hosting was not possible in local machine.
So to return something to the front end, i sent some part of the context as a return.

Comment (tenzin3) [reply backend/backend_main.py]: code has been changed to get response from chatgpt 4 turbo now

File: backend/backend_main.py [hunk 4]
```diff
@@ -100,11 +101,31 @@ def generate_answer(question, num_of_context=5)->str:
...
+def get_chatgpt_response(api_key, prompt):
+    url = "https://api.openai.com/v1/chat/completions"
+    headers = {
+        "Content-Type": "application/json",
+        "Authorization": f"Bearer {api_key}"
+    }
+    data = {
+        "model": "gpt-4-turbo", 
+        "messages": [
+            {"role": "system", "content": "You are a helpful assistant."},
```
Comment (teny19) [inline backend/backend_main.py]: Replace the line "You are a helpful assistant." with "You are his holiness the 14th Dalai Lama." and remove it from the prompt.

File: backend/backend_main.py [hunk 5]
```diff
@@ -0,0 +1,79 @@
...
+    You are his holiness the 14th Dalai Lama.
+    
+    Follow these guidelines when answering the questions:
+    
+    - Answer the question based on the given contexts (some of which might be irrelevant).
+    - Be concise and precise.
+    - Answer directly, without adding any extra words.
+    - Be careful of the language, ensuring it is respectful and appropriate.
+    - If you do not have a proper answer from the context, respond with "I dont have enough data to provide an answer."
+    - Do not give a response longer than 3000 tokens.
```
Comment (teny19) [inline backend/backend_main.py]: Replace this sentence " You are his holiness the 14th Dalai Lama." and add "Your task is to respond to the question based on the provided context.".

**Pull request merged**